### PR TITLE
Randy modified changes

### PR DESCRIPTION
**Pull request by randyclee**
#build

### CODE REVIEW

**Comment by randyclee** [comment]
Fixes: https://github.com/OpenLiberty/open-liberty/issues/13681

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java [hunk 1]
```diff
@@ -45,6 +47,7 @@
     protected static RemoteFile consoleLog;
 
     protected static final int CONN_TIMEOUT = 10;
+    protected static final int threadsToRun = 5;
```
Comment (donbourne) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java:50]: looks like you are defining threadsToRun here and in the testPrintStackTraceMultipleThreads method

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java [hunk 2]
```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
...
+        Logger logger = Logger.getLogger("com.ibm.ws.logging.stackjoiner");
+        logger.log(Level.INFO, ">>>>> Create threads");
+
+        for (int i = 0; i < threadsToRun; i++) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                      messagesLog = server.getDefaultLogFile();
+                      server.setMarkToEndOfLog(messagesLog);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java:135]: here you're setting the mark on the messages log, but then later in this method, on line 153, you are looking for the log content in the console log.

Comment (donbourne) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java:135]: you probably shouldn't be doing this from each thread... would make more sense to do this mark before you spawn the threads.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java [hunk 3]
```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
     }
 
     @Test
-    public void testPrintStackTraceMultipleThreads() {
-        // TODO: Test printStackTrace in multiple threads and ensure that messages are not leaking from one stack trace to another
+    public void testPrintStackTraceMultipleThreads() throws Exception {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java:123]: this test needs more than 3 (or 5) threads to do one hit each to check for concurrency problems.  I'd suggest you have 5 threads and make each run in a loop making, say, 1000 hits each... then in your asserts you can check that you got expected number of events, and check each of the 5000 events to make sure they all have the same number of lines in their stack trace.  you could also check to make sure you have exactly 1000 messages from each thread.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java [hunk 4]
```diff
@@ -117,7 +120,39 @@ public void testPrintStackTraceOverrideReflection() throws Exception {
...
+
+                    } catch (Exception e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    }
+                }
+            }).start();
+            logger.log(Level.INFO, "Thread-" + (i + 1) + " started.");
+            try {
+              Thread.sleep(500);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/StackTraceJoinerTest.java:146]: you'd be better off keeping a reference to each of the threads you start and using Thread.join to wait for each of the threads to complete.

**Review by donbourne — CHANGES_REQUESTED** [review]

**Pull request closed without merging**